### PR TITLE
Source Maps for Concatenated Files

### PR DESCRIPTION
**Pull request by thomas-huston-zocdoc**
@ZocDoc/polaris-devs 

Changes
--
- Generate source maps for concatenated files. This way when we serve up the combined bundle file to the browser in debug mode when working on a require bundle, you'll still be able to inspect the individual files in the dev tools, so it should be entirely transparent.

### CODE REVIEW

File: src/concat/source-map-combiner/combiner.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
+var Generator = require('./generator');
+var getMappings = require('./get-mappings');
+var convert = require('convert-source-map');
+
+function Combiner() {
```
Comment (thomas-huston-zocdoc) [inline src/concat/source-map-combiner/combiner.js]: I've search around for the past few weeks for an existing npm module that does this, and couldn't find one that fits our needs. There were a bunch that were close, but didn't quite to the trick. This is just a simple wrapper around calls to the actual `source-map` library that allows us to concatenate source maps from multiple files into a single source map.

**Comment by thomas-huston-zocdoc** [comment]
As an example, I flipped on the `require` option on `sg.patient-beta.js.bundle`.

In the source, we're just serving up the single unminified combined bundle file:
image

But when debugging, you can navigate to individual files being referenced in their original, uncompiled form:
image
image

And you can debug line by line in those files:
image

File: src/bundler.js [hunk 2]
```diff
@@ -249,22 +249,29 @@ function processJsBundle(options, jsBundle, bundleDir, jsFiles, bundleName, cb)
...
+                fileType: file.type.JS,
+                sourceMap: options.sourcemaps
             });
 
         var afterBundle = function () {
             var minFileName = bundleFileUtility.getMinFileName(bundleName, bundleName, options);
 
-            bundleStatsCollector.AddFileHash(bundleName, allMinJs);
+            bundleStatsCollector.AddFileHash(bundleName, allMinJs.code);
+
```
Comment (Laura-Johannet-ZocDoc) [inline src/bundler.js]: why did this change from allMinJs to allMinJs.code (just curious)

File: src/concat/concat-files.js [hunk 3]
```diff
@@ -26,9 +36,36 @@ function concat(options) {
...
+                line: offset
+            });
+            offset += newlinesIn(file.code) + 1;
+        }
+
     });
 
-    return code.join('');
+    result = {
+        code: code.join('')
```
Comment (Laura-Johannet-ZocDoc) [inline src/concat/concat-files.js]: oh should have just read farther, i take back my question

**Pull request merged**